File: packages/native/macos-native/macos/FRNSliderManager.m
#import <React/RCTComponent.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(FRNSliderManager, RCTViewManager)

RCT_REMAP_VIEW_PROPERTY(value, doubleValue, double)
RCT_REMAP_VIEW_PROPERTY(defaultValue, doubleValue, double)
RCT_REMAP_VIEW_PROPERTY(minimumValue, minValue, double)
RCT_REMAP_VIEW_PROPERTY(maximumValue, maxValue, double)
RCT_EXPORT_VIEW_PROPERTY(numberOfTickMarks, NSInteger)
RCT_CUSTOM_VIEW_PROPERTY(continuous, BOOL, NSSlider) {
  [view setContinuous:json ? [RCTConvert BOOL:json] : YES];
}
RCT_CUSTOM_VIEW_PROPERTY(disabled, BOOL, NSSlider) {
  [view setEnabled:![RCTConvert BOOL:json]];
}
RCT_REMAP_VIEW_PROPERTY(tooltip, toolTip, NSString)
RCT_EXPORT_VIEW_PROPERTY(onValueChange, RCTBubblingEventBlock)

@end

File: packages/native/macos-native/macos/FRNSliderManager.swift
import AppKit
import Foundation
#if USE_REACT_AS_MODULE
import React
#endif // USE_REACT_AS_MODULE

@objc(FRNSliderManager)
class FRNSliderManager: RCTViewManager {

  override func view() -> NSView! {
    return FRNSlider()
  }

  override class func requiresMainQueueSetup() -> Bool {
    return true
  }
}
